### PR TITLE
Deprecate scheduler predicate and priority factory registration and associated types

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removes predicate and priority factory registration, specifically:
1) Removes all predicate registration in register_predicates.go
2) Removes all priority registration in register_priorities.go
3) Removes the types: FitPredicateFactory, PriorityFunctionFactory, PriorityConfigFactory

All predicates and priorities are now statically specified in a couple of maps, those maps will be removed in a followup PR and replaced with direct access to the [ConfigProducerRegistry](https://github.com/kubernetes/kubernetes/blob/bd1195c28e22bbdcb95d718a31186581dc0a5f5f/pkg/scheduler/framework/plugins/registry.go#L119)

Note that all removed unit tests have equivalent ones in other places.

**Which issue(s) this PR fixes**:
Part of #86162

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @draveness

### CODE REVIEW

File: pkg/scheduler/algorithm_factory.go [hunk 1]
```diff
@@ -51,45 +51,68 @@ type AlgorithmFactoryArgs struct {
...
 var (
-	schedulerFactoryMutex sync.RWMutex
-
-	// maps that hold registered algorithm types
-	// TODO(Huang-Wei): remove this.
-	fitPredicateMap        = make(map[string]FitPredicateFactory)
-	mandatoryFitPredicates = sets.NewString()
-	priorityFunctionMap    = make(map[string]PriorityConfigFactory)
-	algorithmProviderMap   = make(map[string]AlgorithmProviderConfig)
+	// predicate keys supported for backward compatibility with v1.Policy.
```
Comment (ahg-g) [inline pkg/scheduler/algorithm_factory.go]: NOTE: the following maps replace the explicit registration in the register_predicates.go and register_priorities.go. This is temporary, a follow up PR will remove those and directly use ConfigProducerRegistry defined in framework/plugins

**Review by draveness — COMMENTED** [review]
/lgtm

Generally looks good, thanks

**Review by zouyee — APPROVED** [review]

**Comment by hectorj2f** [comment]
@ahg-g do you suggest any alternative to replace the predicates from the cluster-autoscaler https://github.com/kubernetes/autoscaler/blob/36fe4949c6f07ee0b80b0cf6b53849b3f90607f6/cluster-autoscaler/simulator/predicates.go#L64?

**Pull request merged**